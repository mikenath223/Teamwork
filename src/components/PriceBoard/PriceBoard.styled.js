import styled from 'styled-components';
import { breakpoints } from 'styles/global';

export const PriceSection = styled.section`
  display: flex;
  justify-content: center;
  margin-bottom: 100px;

  @media ${breakpoints.md} {
    flex-direction: column;
    align-items: center;
  }
`;

export const PriceDetails = styled.article`
  background-color: ${({ theme }) => theme.colors.papaya};
  text-align: center;
  box-shadow: 0 2px 2px silver;
  width: 200px;
  padding: 20px 0;
  margin-right: 40px;

  ul {
    padding: 0;
    list-style: none;
    margin-top: 10px;
  }

  @media ${breakpoints.md} {
    margin: 40px 0 0;
  }
`;

export const PriceTag = styled.h2`
  font-weight: ${({ theme }) => theme.fontWeight.bold};
`;

export const Text = styled.li`
  font-size: ${({ theme }) => theme.textSizes.small};
  font-weight: ${({ theme }) => theme.fontWeight.medium};
  line-height: 30px;
`;

export const PriceFeature = styled.h3`
  font-size: ${({ theme }) => theme.textSizes.large};
  font-weight: ${({ theme }) => theme.fontWeight.bold};
  text-transform: uppercase;
  margin-top: 25px;
`;