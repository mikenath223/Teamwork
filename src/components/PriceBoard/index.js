import React from 'react';
import {
  PriceSection,
  PriceDetails,
  PriceTag,
  PriceFeature,
  Text
} from './PriceBoard.styled';

const PriceBoard = () => (
  <PriceSection>
    <PriceDetails>
      <PriceTag>$0</PriceTag>
      <PriceFeature>Basic</PriceFeature>
      <ul>
        <Text><b>✓</b> Unlimited Chats</Text>
        <Text><b>✓</b> Unlimited Events</Text>
      </ul>
    </PriceDetails>
    <PriceDetails>
      <PriceTag>$40</PriceTag>
      <PriceFeature>Premium</PriceFeature>
      <ul>
        <Text><b>✓</b> Unlimited Chats</Text>
        <Text><b>✓</b> Unlimited Events</Text>
        <Text><b>✓</b> Custom File/Permissions</Text>
        <Text><b>✓</b> Invite-only Private Projects</Text>
      </ul>
    </PriceDetails>
    <PriceDetails>
      <PriceTag>$50</PriceTag>
      <PriceFeature>Ultimate</PriceFeature>
      <ul>
        <Text><b>✓</b> Unlimited Chats</Text>
        <Text><b>✓</b> Unlimited Events</Text>
        <Text><b>✓</b> Custom File/Permissions</Text>
        <Text><b>✓</b> Invite-only Private Projects</Text>
        <Text><b>✓</b> Unlimited Private Projects</Text>
      </ul>
    </PriceDetails>
  </PriceSection>
)

export default PriceBoard;