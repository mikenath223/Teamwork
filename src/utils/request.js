import { auth, generateUserDocument, fire } from './firebaseConfig';
import db from './firebaseConfig';

/*
/ Util function to signup new user 
/ with email and password
*/
export const createUserWithEmailAndPassword = async (email, password, displayName, setError) => {
  try {
    const { user = {} } = await auth.createUserWithEmailAndPassword(email, password)
    await generateUserDocument(user, { displayName });
  } catch (error) {
    setError(error.message);
  }
};

/*
/ Util function to signIn user 
/ with email and password
*/
export const signInUserWithEmailAndPassword = (email, password, setError) => {
  auth.signInWithEmailAndPassword(email, password).catch(() => {
    setError("Error signing in with password and email!");
  });
}

/*
/ Util function to reset user password
*/
export const resetUserPassword = (email, setEmailHasBeenSent, setError) => {
  auth.sendPasswordResetEmail(email)
    .then(() => {
      setEmailHasBeenSent(true);
    })
    .catch(() => {
      setError("Error resetting password");
    });
}

/*
/ Util function to get users
*/
export const getUsers = async () => {
  try {
    let querySnapshot = [];
    await db.collection('users').get().then(query =>
      query.forEach(doc => querySnapshot.push({ id: doc.id, data: doc.data() })))
    return querySnapshot
  } catch (error) {
    console.log('Failed to retrieve users', JSON.stringify(error));
    throw new Error(error);
  }
};

/*
/ Util function to get chats
*/
export const getChats = async (recieverId, currentUserId, runIt) => {
  try {
    await new Promise(function (resolve) {
      return db.collection("chat").doc(recieverId)
        .onSnapshot(function (doc) {
          const chats = doc.data();
          if (chats.data) {
            const filteredChats = chats.data.filter(item =>
              item.recieverId === currentUserId);
            resolve(runIt(filteredChats));
          };
        });
    })
  } catch (error) {
    console.log('Chat uplink ended', JSON.stringify(error));
    throw new Error(error);
  }
}

/*
/ Util function to send chats
*/
export const sendFirestoreChat = (message, senderId, recieverId) => {
  try {
    db.collection('chat').doc(senderId).get().then((doc) => {
      if (doc.exists) {
        db.collection('chat').doc(senderId).update({
          data: fire.firestore.FieldValue.arrayUnion({
            message, recieverId, date: new Date().toLocaleString()
          })
        });
      } else {
        db.collection('chat').doc(senderId).set({
          data: [{
            message, recieverId, date: new Date().toLocaleString()
          }]
        });
      }
    })
  } catch (error) {
    console.log('Failed to send chat', JSON.stringify(error));
    throw new Error(error);
  }
};

/*
/ Util function to group chats
*/
export const runDispatch = (moment, dispatch, setChats,
  chats = [], count = 0) => (input) => {

    if (count >= 2) {
      count = 0;
      chats = [];
    }
    chats.push(...input);
    if (chats.length) {
      chats = chats.sort((a, b) =>
        new Date(a.date) - new Date(b.date))
      chats = chats.map(item => ({
        date: moment(item.date).fromNow(),
        message: item.message,
        recieverId: item.recieverId
      }));
      dispatch(setChats(chats));
    };
    count += 1;
  };

/*
/ Util function to save meetings
*/
export const sendFirebaseEvent = (event, currentUserId) => {
  db.collection('meetings').doc(currentUserId).get().then((doc) => {
    if (doc.exists) {
      db.collection('meetings').doc(currentUserId).update({
        data: fire.firestore.FieldValue.arrayUnion(event)
      });
    } else {
      db.collection('meetings').doc(currentUserId).set({
        data: [event]
      });
    }
  })
}