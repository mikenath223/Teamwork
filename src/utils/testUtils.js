import React from 'react';
import { ThemeProvider } from 'styled-components';
import { theme } from 'styles/global';
import { render } from '@testing-library/react';


export const withThemeProvider = (component) =>
  render(<ThemeProvider theme={theme}>{component}</ThemeProvider>);