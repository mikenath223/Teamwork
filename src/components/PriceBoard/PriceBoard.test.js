import React from 'react';
import 'jest-styled-components';
import { withThemeProvider } from 'utils/testUtils';
import PriceBoard from './index';

describe('PriceBoard', function () {
  it('should match snapshot', () => {
    const { container } = withThemeProvider(<PriceBoard />);

    expect(container).toMatchSnapshot();
  })
})