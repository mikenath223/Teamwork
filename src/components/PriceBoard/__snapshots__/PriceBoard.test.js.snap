// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PriceBoard should match snapshot 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  margin-bottom: 100px;
}

.c1 {
  background-color: #FCFCFC;
  text-align: center;
  box-shadow: 0 2px 2px silver;
  width: 200px;
  padding: 20px 0;
  margin-right: 40px;
}

.c1 ul {
  text-align: left;
  list-style: none;
  padding: 0 5px;
  margin-top: 10px;
}

.c2 {
  font-weight: 700;
}

.c4 {
  font-size: 12px;
  font-weight: 500;
  padding-left: 16px;
  line-height: 30px;
}

.c3 {
  font-size: 18px;
  font-weight: 700;
  text-transform: uppercase;
  margin-top: 25px;
}

@media (max-width:768px) {
  .c0 {
    -webkit-flex-direction: column;
    -ms-flex-direction: column;
    flex-direction: column;
    -webkit-align-items: center;
    -webkit-box-align: center;
    -ms-flex-align: center;
    align-items: center;
  }
}

@media (max-width:768px) {
  .c1 {
    margin: 40px 0 0;
  }
}

<div>
  <section
    class="c0"
  >
    <article
      class="c1"
    >
      <h2
        class="c2"
      >
        $35
      </h2>
      <h3
        class="c3"
      >
        Basic
      </h3>
      <ul>
        <li
          class="c4"
        >
          <b>
            ✓
          </b>
           Unlimited Projects
        </li>
        <li
          class="c4"
        >
          <b>
            ✓
          </b>
           Unlimited Version History
        </li>
      </ul>
    </article>
    <article
      class="c1"
    >
      <h2
        class="c2"
      >
        $40
      </h2>
      <h3
        class="c3"
      >
        Premium
      </h3>
      <ul>
        <li
          class="c4"
        >
          <b>
            ✓
          </b>
           Unlimited Projects
        </li>
        <li
          class="c4"
        >
          <b>
            ✓
          </b>
           Unlimited Version History
        </li>
        <li
          class="c4"
        >
          <b>
            ✓
          </b>
           Custom File/Permissions
        </li>
        <li
          class="c4"
        >
          <b>
            ✓
          </b>
           Invite-only Private Projects
        </li>
      </ul>
    </article>
    <article
      class="c1"
    >
      <h2
        class="c2"
      >
        $50
      </h2>
      <h3
        class="c3"
      >
        Ultimate
      </h3>
      <ul>
        <li
          class="c4"
        >
          <b>
            ✓
          </b>
           Unlimited Projects
        </li>
        <li
          class="c4"
        >
          <b>
            ✓
          </b>
           Unlimited Version History
        </li>
        <li
          class="c4"
        >
          <b>
            ✓
          </b>
           Custom File/Permissions
        </li>
        <li
          class="c4"
        >
          <b>
            ✓
          </b>
           Invite-only Private Projects
        </li>
        <li
          class="c4"
        >
          <b>
            ✓
          </b>
           Unlimited Private Projects
        </li>
      </ul>
    </article>
  </section>
</div>
`;
